src/persist-client/src/inspect.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! CLI introspection tools for persist
+
+use crate::r#impl::state::ProtoStateRollup;
+use crate::{Metrics, ShardId};
+use mz_ore::metrics::MetricsRegistry;
+use mz_persist::cfg::ConsensusConfig;
+use mz_persist::location::SeqNo;
+use prost::Message;
+use serde_json::json;
+
+/// Prints the current state of a given shard as JSON
+pub async fn print_current_state(
```

Comment:
having any of this code live in `persist-client` directly is a little regrettable, but the `examples` target can only use the public interface of the crate, so it does not have access to `ProtoStateRollup`. 

I'm not sure if there's an easy and more elegant crate/module thing we can do to better organize the code

Reply:
i'm okay with it

at some point, I'd like to extract some sort of API trait between persist-client and the actual impl bits, then most code using persist only has to compile against persist-client and only the various bin targets point at the impl. this would continue to live in the impl bits and at that point I think it would feel much better